connector/src/main/scala/com/google/cloud/spark/bigquery/ArrowBinaryIterator.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2018 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.spark.bigquery
+
+import java.io.{ByteArrayInputStream, InputStream}
+
+import com.google.cloud.spark.bigquery.converters.ArrowSchemaConverter
+import com.google.protobuf.ByteString
+import org.apache.arrow.vector.VectorSchemaRoot
+import org.apache.arrow.vector.ipc.{ArrowReader, ArrowStreamReader}
+import org.apache.spark.sql.catalyst.InternalRow
+import org.apache.spark.sql.execution.arrow.ArrowUtils
+import org.apache.spark.sql.vectorized.{ColumnVector, ColumnarBatch}
+
+import collection.JavaConverters._
+
+/**
+ * An iterator for scanning over rows serialized in Arrow format
+ * @param columnsInOrder Sequence of columns in the schema
+ * @param schema Schema in avro format
+ * @param rowsInBytes Rows serialized in binary format for Arrow
+ */
+class ArrowBinaryIterator(columnsInOrder: Seq[String],
```

Comment:
Make both ArrowBinaryIterator and AvroBinaryIterator inherit from a defined interface.